modules/SofaUserInteraction/RayNewProximityIntersection.cpp

```diff
@@ -231,17 +232,16 @@ int RayNewProximityIntersection::computeIntersection(Ray& rRay, RigidSphere& rSp
 
 
 //			const SReal contactDist = fHitFraction;
-            contacts->resize(contacts->size()+1);
-            DetectionOutput *detection = &*(contacts->end()-1);
-
-            detection->elem = std::pair<core::CollisionElementIterator, core::CollisionElementIterator>(rRay, rSphere);
-            detection->point[1] = v3ContactPoint;
-            detection->point[0] = v3ContactPoint;
-            detection->normal = v3Normal;
-            detection->value = fHitFraction;
-            detection->value -= fHitFraction;
 
+            DetectionOutput detection;
+            detection.elem = std::pair<core::CollisionElementIterator, core::CollisionElementIterator>(rRay, rSphere);
+            detection.point[1] = v3ContactPoint;
+            detection.point[0] = v3ContactPoint;
+            detection.normal = v3Normal;
+            detection.value = fHitFraction;
+            detection.value -= fHitFraction;
```

Comment:
In every file, the `DetectionOutput`is initialized the same way. Can't we get ctor that does that ?